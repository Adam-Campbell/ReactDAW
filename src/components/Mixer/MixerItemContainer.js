import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as ActionCreators from '../../actions';
import PropTypes from 'prop-types';
import MixerItem from './MixerItem';

export class MixerItemContainer extends Component {
    
    get track() {
        return this.props.channels.find(channel => channel.id === this.props.trackId);
    }

    handleVolumeChange = (e) => {
        this.props.updateChannelVolume(this.props.trackId, parseFloat(e.target.value));
    }

    handlePanChange = (e) => {
        this.props.updateChannelPan(this.props.trackId, parseFloat(e.target.value));
    }

    handleMuteButtonClick = () => {
        if (this.track.isMuted) {
            this.props.unmuteChannel(this.props.trackId);
        } else {
            this.props.muteChannel(this.props.trackId);
        }
    }

    handleSoloButtonClick = () => {
        if (this.track.isSolo) {
            this.props.unsoloChannel(this.props.trackId);
        } else {
            this.props.soloChannel(this.props.trackId);
        }
    }

    render() {
        return <MixerItem 
            volume={this.track.volume}
            handleVolumeChange={this.handleVolumeChange}
            trackId={this.props.trackId}
            handleMuteButtonClick={this.handleMuteButtonClick}
            handleSoloButtonClick={this.handleSoloButtonClick}
            pan={this.track.pan}
            handlePanChange={this.handlePanChange}
            name={this.track.name}
            isMuted={this.track.isMuted}
            isSolo={this.track.isSolo}
        />
    }
}

MixerItemContainer.propTypes = {
    trackId: PropTypes.string.isRequired
}

const mapStateToProps = state => ({
    channels: state.main.present.channels
});



export default connect(
    mapStateToProps,
    {
        updateChannelVolume: ActionCreators.updateChannelVolume,
        muteChannel: ActionCreators.muteChannel,
        unmuteChannel: ActionCreators.unmuteChannel,
        soloChannel: ActionCreators.soloChannel,
        unsoloChannel: ActionCreators.unsoloChannel,
        updateChannelPan: ActionCreators.updateChannelPan
    }
)(MixerItemContainer);
