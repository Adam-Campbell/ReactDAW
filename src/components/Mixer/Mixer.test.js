import React from 'react';
import { shallow } from 'enzyme';
import { Mixer } from './';
import { MixerItemContainer } from './MixerItemContainer';

const channelsState = [
    {
      id: '2812513466104156',
      name: 'Channel 1',
      color: '#d86597',
      instrumentId: '8516328939937592',
      effectIds: [],
      sectionIds: [],
      volume: 0,
      isMuted: false,
      isSolo: false,
      pan: 0
    },
    {
      id: '3880564538936934',
      name: 'Channel 2',
      color: '#d86597',
      instrumentId: '1050717713415613',
      effectIds: [],
      sectionIds: [],
      volume: 0,
      isMuted: false,
      isSolo: false,
      pan: 0
    },
    {
      id: '7214036741073003',
      name: 'Channel 3',
      color: '#d86597',
      instrumentId: '4150723633346982',
      effectIds: [],
      sectionIds: [],
      volume: 0,
      isMuted: false,
      isSolo: false,
      pan: 0
    }
];


test('renders correctly', () => {
    const component = shallow(
        <Mixer 
            channels={channelsState}
        />
    );
    expect(component).toMatchSnapshot();
});



const sampleState = {
    playerInfo: {
      isPlaying: false,
      isMuted: false,
      volume: 0,
      bpm: 120
    },
    activeWindows: [],
    channels: [
      {
        id: '2812513466104156',
        name: 'Channel 1',
        color: '#d86597',
        instrumentId: '8516328939937592',
        effectIds: [],
        sectionIds: [],
        volume: 0,
        isMuted: false,
        isSolo: false,
        pan: 0
      },
      {
        id: '3880564538936934',
        name: 'Channel 2',
        color: '#d86597',
        instrumentId: '1050717713415613',
        effectIds: [],
        sectionIds: [],
        volume: 0,
        isMuted: false,
        isSolo: false,
        pan: 0
      },
      {
        id: '7214036741073003',
        name: 'Channel 3',
        color: '#d86597',
        instrumentId: '4150723633346982',
        effectIds: [],
        sectionIds: [],
        volume: 0,
        isMuted: false,
        isSolo: false,
        pan: 0
      }
    ],
    sections: {},
    instruments: {
      '8516328939937592': {
        id: '8516328939937592',
        channelId: '2812513466104156',
        type: 'synth',
        instrumentData: {
          envelope: {
            attack: 0.005,
            attackCurve: 'linear',
            decay: 0.1,
            release: 1,
            releaseCurve: 'exponential',
            sustain: 0.3
          },
          oscillator: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'triangle',
            volume: 0
          },
          portamento: 0,
          volume: 0
        }
      },
      '1050717713415613': {
        id: '1050717713415613',
        channelId: '3880564538936934',
        type: 'amSynth',
        instrumentData: {
          detune: 0,
          envelope: {
            attack: 0.01,
            attackCurve: 'linear',
            decay: 0.01,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
          },
          harmonicity: 3,
          modulation: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'square',
            volume: 0
          },
          modulationEnvelope: {
            attack: 0.5,
            attackCurve: 'linear',
            decay: 0,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
          },
          oscillator: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'triangle',
            volume: 0
          },
          portamento: 0,
          volume: 0
        }
      },
      '4150723633346982': {
        id: '4150723633346982',
        channelId: '7214036741073003',
        type: 'fmSynth',
        instrumentData: {
          detune: 0,
          envelope: {
            attack: 0.01,
            attackCurve: 'linear',
            decay: 0.01,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
          },
          harmonicity: 3,
          modulation: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'square',
            volume: 0
          },
          modulationEnvelope: {
            attack: 0.5,
            attackCurve: 'linear',
            decay: 0,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
          },
          modulationIndex: 10,
          oscillator: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'triangle',
            volume: 0
          },
          portamento: 0,
          volume: 0
        }
      }
    },
    effects: {}
  }