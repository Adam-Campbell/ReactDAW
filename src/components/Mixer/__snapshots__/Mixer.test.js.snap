// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mixer__container"
>
  <div
    className="mixer__channel-controls-container"
  >
    <Connect(MixerItemContainer)
      key="2812513466104156"
      trackId="2812513466104156"
    />
    <Connect(MixerItemContainer)
      key="3880564538936934"
      trackId="3880564538936934"
    />
    <Connect(MixerItemContainer)
      key="7214036741073003"
      trackId="7214036741073003"
    />
  </div>
  <Connect(MasterSliderContainer) />
</div>
`;
